docs/resources/apig_signature.md

```diff
@@ -0,0 +1,122 @@
+
+Manages a signature resource within FlexibleEngine.
+
+## Example Usage
+
+### Create a signature of the HMAC type
+
+```hcl
+variable "instance_id" {}
+variable "signature_name" {}
+variable "signature_key" {}
+variable "signature_secret" {}
+
+resource "flexibleengine_apig_signature" "test" {
+  instance_id = var.instance_id
+  name        = var.signature_name
+  type        = "hmac"
+  key         = var.signature_key
+  secret      = var.signature_secret
+}
+```
+
+### Create a signature and automatically generate key and secret
+
+```hcl
+variable "instance_id" {}
+variable "signature_name" {}
+
+resource "flexibleengine_apig_signature" "test" {
+  instance_id = var.instance_id
+  name        = var.signature_name
+  type        = "hmac"
+}
+```
+
+### Create a signature of the AES type
+
+```hcl
+variable "instance_id" {}
+variable "signature_name" {}
+variable "signature_key" {}
+variable "signature_secret" {}
+
+resource "flexibleengine_apig_signature" "test" {
+  instance_id = var.instance_id
+  name        = var.signature_name
+  type        = "aes"
+  algorithm   = "aes-128-cfb"
+  key         = var.signature_key
+  secret      = var.signature_secret
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region where the signature is located.  
+  If omitted, the provider-level region will be used. Changing this will create a new resource.
+
+* `instance_id` - (Required, String, ForceNew) Specifies the ID of the dedicated instance to which the signature
+  belongs.  
+  Changing this will create a new resource.
+
+* `name` - (Required, String) Specifies the signature name.  
+  The valid length is limited from `3` to `64`, only English letters, Chinese characters, digits and underscores (_) are
+  allowed. The name must start with an English letter or Chinese character.
+
+* `type` - (Required, String) Specifies the type of signature.  
+  The valid values are as follows:
+  + **basic**: Basic auth type.
+  + **hmac**: HMAC type.
+  + **aes**: AES type
+
+  Changing this will create a new resource.
+
+* `key` - (Optional, String) Specifies the signature key.  
+  + For `basic` type: The value contains `4` to `32` characters, including letters, digits, underscores (_) and
+    hyphens (-). It must start with a letter.
+  + For `hmac` type: The value contains `8` to `32` characters, including letters, digits, underscores (_) and
+    hyphens (-). It must start with a letter or digit.
+  + For `aes` type: The value contains `16` characters if the `aes-128-cfb` algorithm is used, or `32` characters if the
+    `aes-256-cfb` algorithm is used. Only letters, digits, and special characters (`_-!@#$%+/=`) are allowed.
+    It must start with a letter, digit, plus sign (+), or slash (/).
+
+  If not specified, the key will automatically generated. The auto-generation is only supported on first creation.  
+  Changing this will create a new resource.
```

Comment:
This field is not ForceNew, please remove `Changing this will create a new resource.`

Reply:
![image](https://github.com/FlexibleEngineCloud/terraform-provider-flexibleengine/assets/145660317/4b0fc57b-b0cc-4ec2-aec9-cbf48170c567)
done